server/providers/challenges/rdeploy-blob/index.ts

```diff
@@ -17,30 +31,89 @@ class RDeployBlobProvider extends EventEmitter implements Provider {
   private _interval: NodeJS.Timeout
   private challenges: Challenge[]
 
+  private ready = false
+  private downloadMap: Map<string, string>
+
   constructor (options: RDeployBlobProviderOptions) {
     super()
     options = {
       updateInterval: 60 * 1000,
-      useGlobalRDeployDirectory: true,
-      // rDeployDirectory: ''
       ...options
     }
 
     this._updateInterval = options.updateInterval
-    this._rDeployDirectory = options.useGlobalRDeployDirectory
-      ? config.rDeployDirectory
-      : options.rDeployDirectory
+    this._rDeployDirectory = path.join(__dirname, '../../../../', options.rDeployDirectory)
     this._interval = setInterval(() => this._update(), this._updateInterval)
-    this._update()
+
+    this.downloadMap = new Map()
+
+    const fileDir = path.join(this._rDeployDirectory, options.rDeployFiles)
+    fs.readdir(fileDir)
+      .then(files => {
+        return Promise.all(
+          files.map(
+            file => {
```

Comment:
We should filter to remove files which start with `.`, in case `.DS_Store` or similar gets in the dir somehow.

Reply:
Does rDeploy never create dotfiles? This gets complicated when there are dotfiles that we actually want to serve (e.g. challenge where a `.bash_history` is involved)

Reply:
If it's not referenced in a problem, there's no way to actually download the file. They'd have to guess the hash, which is essentially the same as guessing the contents of the file.